# HG changeset patch
# User Rik <rik@octave.org>
# Date 1542211480 28800
#      Wed Nov 14 08:04:40 2018 -0800
# Node ID 01da3729016ad83fe46060c067e11db77bb71ff5
# Parent  edc02d5eeac6a22d05b206caa9421691f0e313f8
strsplit.m: Simplify input validation test for multi-line inputs in cset edc02d5eeac6.

* strsplit.m: Use rows (str) > 1 to test for multi-line char vectors.

diff --git a/scripts/strings/strsplit.m b/scripts/strings/strsplit.m
--- a/scripts/strings/strsplit.m
+++ b/scripts/strings/strsplit.m
@@ -172,17 +172,17 @@ function [cstr, matches] = strsplit (str
       del = {" ", "\f", "\n", "\r", "\t", "\v"};
     else
       del = '\s';
     endif
   endif
 
   if (! ischar (str) || (! ischar (del) && ! iscellstr (del)))
     error ("strsplit: S and DEL must be string values");
-  elseif (numel (str) != columns (str))
+  elseif (rows (str) > 1)
     error ("strsplit: S must be a char row vector")
   elseif (! isscalar (args.collapsedelimiters))
     error ("strsplit: COLLAPSEDELIMITERS must be a scalar value");
   endif
 
   if (strncmpi (args.delimitertype, "simple", length_deltype))
     is_simple = true;
   elseif (strncmpi (args.delimitertype, "regularexpression", length_deltype))
@@ -304,11 +304,11 @@ endfunction
 
 %!assert <*47403> (strsplit ('xxx+yyy', '+'), {"xxx", "yyy"})
 
 ## Test input validation
 %!error strsplit ()
 %!error strsplit ("abc", "b", true, 4)
 %!error <invalid parameter name, 'foo'> strsplit ("abc", "b", "foo", "true")
 %!error <S and DEL must be string values> strsplit (123, "b")
+%!error <S must be a char row vector> strsplit (["abc"; "xyz"])
 %!error <COLLAPSEDELIMITERS must be a scalar value> strsplit ("abc", "def", "collapsedelimiters", ones (3,3))
 %!error <Invalid DELIMITERTYPE> strsplit ("abc", "b", "delimitertype", "foobar")
-%!error <S must be a char row vector> strsplit (reshape ("axbycz", [2 3]))
